pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
```

Comment:
Let's leave lease record data out of initial implementation. Annotations can be used and we can add to decide it later. @rrati I think the string represenation is for HeldBy. The representation of ownership has come up in a bunch of issues (mostly with controllers owning certain api objects) but I'm not sure it applies here since the owner will not always be an api object.